rust/src/push/mod.rs

```diff
@@ -280,6 +280,37 @@ impl<'source> FromPyObject<'source> for SimpleJsonValue {
     }
 }
 
+/// A JSON values (list, string, int, boolean, or null).
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(untagged)]
+pub enum JsonValue {
+    Array(Vec<SimpleJsonValue>),
+    Value(SimpleJsonValue),
+}
+
+impl<'source> FromPyObject<'source> for JsonValue {
+    fn extract(ob: &'source PyAny) -> PyResult<Self> {
+        if let Ok(l) = <PyList as pyo3::PyTryFrom>::try_from(ob) {
+            if let Ok(a) = l.iter().map(SimpleJsonValue::extract).collect() {
+                Ok(JsonValue::Array(a))
+            } else {
+                // TOOD How to get a good error?
```

Comment:
Should we remove the TODO ? if not, typo :)

Reply:
I'm guessing (without looking closely) that `.collect()` gives you a `Result`. If so, the `if let Ok(a)` discards the Error part of that result before you can do anything with it.

I think you want to either `match` on that Result, or use one of the fancy methods on Result (.map, .map_err?).

Reply:
Thanks @DMRobertson! Switching to `match` made it doable. Thought I had tried that but who knows...